File: aula-01/java/Ex08.java
public class Ex08 {

    public int Produto(int x, int y) {

        if(x < 0) {
            throw new IllegalArgumentException("Intervalo de x inválido");
        }

        if(y < 0) {
            throw new IllegalArgumentException("Intervalo de y inválido");
        }

        int i = 1;
        int potencia = 1;

        while(i <= y) {
            potencia = Produto(potencia, x);
            i = i + 1;
        }

        return potencia;

}

File: aula-01/js/Ex08.js
/**
 * Verifica potencia de x elevado a y utilizando somas
 * @param {Number} x base da potência
 * @param {Number} y expoente da potência
 * @throws {RangeError} caso x e y sejam menores que 0
 * @returns {Number} retorna o valor da potência de x por y
 */
module.exports = potencia;
function potencia(x, y) {

    if (x < 0) {
        throw new RangeError("Intervalo de x inválido");
    }

    if (y < 0) {
        throw new RangeError("Intervalo de y inválido");
    }

    let potencia = 1;
    let i = 1;
    let Produto;
    while (i <= y) {
        potencia = Produto(potencia, x);
        i = i + 1;
    }

    return potencia;
}